### PR TITLE
Changed `invariant` Abstraction to `Heuristic`

### PR DESCRIPTION
**Pull request by epociask**
## Fixes Issue

Fixes #
## Changes proposed
The term `invariant` is rather unorthodox and difficult for people to reason about; especially provided that its a more formal mathematical term. Because of this, we have opted to use a more clean/concise abstraction. Internal conversations polled for the term `Heuristic` to be used.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,6 +117,5 @@ A bootstrap config file is used to define the initial state of the pessimism ser
 ```
 
 
-
-## Spawning an invariant session
-To learn about the currently supported invariants and how to spawn them, please advise the [invariants' documentation](./docs/invariants.md).
+## Spawning an heuristic session
```
Comment (adrain-cb) [inline README.md:120]:
```suggestion
## Spawning a heuristic session
```

**Review by adrain-cb — COMMENTED** [review]
LGTM for the most part. The docs changes in #129 should fix diagrams within gh pages, maybe we can merge that one in first so the pr doesn't contain all the doc additions made in this pr.

File: README.md (hunk 1 above)
Comment (epociask) [reply README.md]: [9708c2f](https://github.com/base-org/pessimism/pull/128/commits/9708c2f03f872a6ef8ca005c09c51d6f3030a6fc)

**Review by adrain-cb — COMMENTED** [review]
Remove changes to everything in ./docs, most of these files have been renamed. If there are any additions can you copy them into the new files?

**Review by adrain-cb — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for adrain-cb @ 2023-08-01 18:12:24 UTC
User cannot review their own commit

**Review by anupsv-cb — APPROVED** [review]

**Pull request merged**